Sources/ToolchainRegistry/ToolchainRegistry.swift

```diff
@@ -219,7 +220,9 @@ package final actor ToolchainRegistry {
     }
 
     let toolchainsAndReasons = toolchainPaths.compactMap {
-      if let toolchain = Toolchain($0.path) {
+      if let resolvedPath = try? $0.path.realpath,
```

Comment:
Same here, should we use `orLog`. I imagine this could help if the user has broken symlinks in their path. Or generally for all uses of `realpath`.